analyzer/emerald/incoming.go

```diff
@@ -344,23 +344,31 @@ func extractRound(b *block.Block, txrs []*sdkClient.TransactionWithResults, logg
 					ERC20Transfer: func(fromEthAddr []byte, toEthAddr []byte, amountU256 []byte) error {
 						amount := &big.Int{}
 						amount.SetBytes(amountU256)
-						if !bytes.Equal(fromEthAddr, common.ZeroEthAddr) {
+						fromZero := bytes.Equal(fromEthAddr, common.ZeroEthAddr)
+						toZero := bytes.Equal(toEthAddr, common.ZeroEthAddr)
+						if !fromZero {
 							fromAddr, err2 := registerRelatedEthAddress(blockData.AddressPreimages, blockTransactionData.RelatedAccountAddresses, fromEthAddr)
 							if err2 != nil {
 								return fmt.Errorf("from: %w", err2)
 							}
 							registerTokenDecrease(blockData.TokenBalanceChanges, eventAddr, fromAddr, amount)
 						}
-						if !bytes.Equal(toEthAddr, common.ZeroEthAddr) {
+						if !toZero {
 							toAddr, err2 := registerRelatedEthAddress(blockData.AddressPreimages, blockTransactionData.RelatedAccountAddresses, toEthAddr)
 							if err2 != nil {
 								return fmt.Errorf("to: %w", err2)
 							}
 							registerTokenIncrease(blockData.TokenBalanceChanges, eventAddr, toAddr, amount)
```

Comment:
In the case of mints/burns, will we `registerTotalSupplyIncrease/Decrease()`?

Reply:
it's not settled, and the total supply dead reckoning isn't implemented yet. if we do, then I suppose this is where we'll put it. current issue is how do we coordinate updates to the total supply in the db? will it be two separate numbers (one dead reckoned and one read from the contract)? how will the API know which one to use?

Reply:
ok let's shelve dead reckoning for now. I expect that the token analyzer will catch up fast, and also that hardly anyone cares about total supply being exactly right.

For the other concerns: I was thinking the two analyzers would use the same output column, and the evm_token analyzer would only overwrite the data if it was obtained at round that is >= the round of dead reckoning. Maybe I'm missing a corner case?

Reply:
sure, let's shelve.

> evm_token analyzer would only overwrite the data if it was obtained at round that is >= the round of dead reckoning

hm yeah, i.e. if the token analyzer is fully caught up. possibly a little fiddly in SQL. we need to join with the round progress table or something. in an UPDATE query :thinking:

Reply:
resolving for merge. added a link from #293

Reply:
> we need to join with the round progress table or something. in an UPDATE query.

I played around with that before I realized we don't have dead reckoning yet, actually :)
You can do something like
```sql
UPDATE tokens
SET ...
WHERE contract_address = ... AND
current_round >= (
	SELECT last_updated_round
	FROM token_analysis_progress
	WHERE contract_address=tokens.contract_address
)
```